bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/discovery/HDPowerViewSddpDiscoveryParticipant.java

```diff
@@ -56,15 +66,21 @@ public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
         final ThingUID thingUID = getThingUID(device);
         if (thingUID != null) {
             try {
+                String label;
+                String serial;
                 int generation = getGeneration(device);
-                String label = generation == 3 //
-                        ? String.format("@text/%s [\"%s\"]", LABEL_KEY_GATEWAY, device.ipAddress)
-                        : String.format("@text/%s [\"%s\", \"%s\"]",
-                                HDPowerViewHubMDNSDiscoveryParticipant.LABEL_KEY_HUB, device.ipAddress, generation);
-
+                if (generation < 3) {
+                    label = String.format("@text/%s [\"%s\", \"%s\"]",
+                            HDPowerViewHubMDNSDiscoveryParticipant.LABEL_KEY_HUB, generation, device.ipAddress);
```

Comment:
Nice catch. Perhaps we should extract this fix to a separate PR with a single-line fix that could also be backported to 4.3.x?

Reply:
Ok